fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1031,16 +1036,31 @@ private void handleKill() throws DdlException {
             handleKillQuery(killStmt.getQueryId());
         } else {
             long id = killStmt.getConnectionId();
-            ConnectContext killCtx = context.getConnectScheduler().getContext(id);
+            ConnectContext killCtx = null;
+            if (isProxy) {
+                final String hostName = context.getProxyHostName();
```

Comment:
How can you guarantee that hostName is not null.

Reply:
The null ConnectContext will be returned if hostName is null
```
    public ConnectContext getContext(String hostname, Integer connectionId) {
        final Map<Integer, ConnectContext> contextMap = connectionMaps.get(hostname);
        if (contextMap == null) {
            return null;
        }
        return contextMap.get(connectionId);
    }
```
then an exception will be reported
```
            if (killCtx == null) {
                ErrorReport.reportDdlException(ErrorCode.ERR_NO_SUCH_THREAD, id);
            }
```